### PR TITLE
added node attributes (customNodeLabels) to agent pools

### PR DESCRIPTION
**Pull request by xtophs**
Do-over of #138

---
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/362)

### CODE REVIEW

File: parts/dcoscustomdata173.t [hunk 1]
```diff
@@ -346,3 +346,7 @@ write_files:
   path: "/opt/azure/containers/provision.sh"
   permissions: "0744"
   owner: "root"
+- path: /var/lib/dcos/mesos-slave-common
```
Comment (colemickens) [inline parts/dcoscustomdata173.t]: this is all it takes? You just put the map into `mesos-slave-common` and it knows to apply those labels to the nodes?

File: pkg/api/convertertoapi.go [hunk 2]
```diff
@@ -271,6 +271,10 @@ func convertVLabsAgentPoolProfile(vlabs *vlabs.AgentPoolProfile, api *AgentPoolP
 	api.VnetSubnetID = vlabs.VnetSubnetID
 	api.Subnet = vlabs.GetSubnet()
 	api.FQDN = vlabs.FQDN
+	api.CustomNodeLabels = map[string]string{}
```
Comment (colemickens) [inline pkg/api/convertertoapi.go]: Does this work? You don't need to `make(map[string]string)` it? I can't remember in Go...

File: parts/dcoscustomdata173.t [hunk 3]
```diff
@@ -346,3 +346,7 @@ write_files:
...
+  content: 'ATTRIBUTES_STR'
```
Comment (colemickens) [inline parts/dcoscustomdata173.t]: How does `ATTRIBUTES_STR` get populated?

File: parts/dcoscustomdata173.t (hunk 1 above)
Comment (JackQuincy) [inline parts/dcoscustomdata173.t]: https://dcos.io/docs/1.7/administration/faq/
this says yes

File: parts/dcoscustomdata173.t (hunk 3 above)
Comment (JackQuincy) [inline parts/dcoscustomdata173.t]: Christoph looks like you dropped some changes in engine.go

**Comment by xtophs** [comment]
Yes. They are coming through shortly … committing, squashing …

Eveything works tho:

[cid:image001.png@01D2944D.8E9AFBC0]

**Comment by JackQuincy** [comment]
Lgtm.

**Pull request merged**